src/server/pfs/s3/driver.go

```diff
@@ -70,25 +73,29 @@ func (d *MasterDriver) listBuckets(pc *client.APIClient, r *http.Request, bucket
 }
 
 func (d *MasterDriver) bucket(pc *client.APIClient, r *http.Request, name string) (*Bucket, error) {
-	branch := "master"
-	var repo string
+	var repo, branch, commit string
+	commit, branch = "master", "master"
 	parts := strings.SplitN(name, ".", 2)
 	if len(parts) == 2 {
-		branch = parts[0]
+		if !uuid.IsUUIDWithoutDashes(parts[0]) {
+			branch = parts[0]
+		}
+		commit = parts[0]
 		repo = parts[1]
 	} else {
 		repo = parts[0]
 	}
 
 	return &Bucket{
 		Repo:   repo,
-		Commit: branch,
+		Branch: branch,
+		Commit: commit,
 		Name:   name,
 	}, nil
 }
 
 func (d *MasterDriver) bucketCapabilities(pc *client.APIClient, r *http.Request, bucket *Bucket) (bucketCapabilities, error) {
-	branchInfo, err := pc.InspectBranch(bucket.Repo, bucket.Commit)
```

Comment:
This broke the `<commit>.<repo>` syntax, because it was trying to inspect a branch but given commit_id instead.